frontend/src/app/modules/grids/widgets/news/news.component.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { NewsDmService } from "core-app/modules/hal/dm-services/news-dm.service";
+import { ComponentFixture, fakeAsync, TestBed, tick, async } from '@angular/core/testing';
+import { WidgetNewsComponent } from './news.component';
+import { DebugElement, NO_ERRORS_SCHEMA } from '@angular/core';
+import { TimezoneService } from 'core-app/components/datetime/timezone.service';
+import { ConfigurationService } from 'core-app/modules/common/config/configuration.service';
+import { States } from 'core-app/components/states.service';
+import { UserDmService } from 'core-app/modules/hal/dm-services/user-dm.service';
+import { HalResourceService } from "core-app/modules/hal/services/hal-resource.service";
+import { HttpClientModule } from "@angular/common/http";
+
+describe('shows news', () => {
+  let app:WidgetNewsComponent;
+  let fixture:ComponentFixture<WidgetNewsComponent>;
+
+
+  let newsStub = {
+    id: 1,
+    title: 'Welcome to your demo project',
+    author: {
+      href: '/api/v3/users/1',
+      name: 'Foo Bar'
+    },
+    summary: 'We are glad you joined. In this module you can communicate project news to your team members.\n',
+    description: {
+      format: 'markdown',
+      raw: 'The actual news',
+      html: '<p>The actual news</p>'
+    },
+    createdAt: '2020-03-26T10:42:14Z',
+    updatedAt: '2020-03-26T10:42:14Z',
+  };
+
+  let newsDmStub = {
+    list: (_params:any) => {
+      return Promise.resolve({ elements: [newsStub] });
+    }
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        WidgetNewsComponent],
+      providers: [
+        TimezoneService,
+        { provide: ConfigurationService, useValue: {} },
+        States,
+        UserDmService,
+        { provide: NewsDmService, useValue: newsDmStub },
+        HalResourceService,
+      ],
+      imports: [HttpClientModule],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(WidgetNewsComponent);
+    app = fixture.debugElement.componentInstance;
+
+  });
+
+  it('should load news from the server', fakeAsync(() => {
+
+    fixture.detectChanges();
+    tick();
+    expect(app.entries.length).toBe(1);
+
```

Comment:
The spec could use some added testing from the DOM. For this there needs to be some changes:

1. You need to call `cdr.detectChanges()` in news.component `setupNews` method to ensure the view is rebuilt.

2. You then need to stub more methods in the `ConfigurationService` because the spec is trying to access methods like `dateFormatPresent`

```
  let configurationServiceStub = {
    isTimezoneSet: () => false,
    dateFormatPresent: () => false,
    timeFormatPresent: () => false
  };
```

and then below in the providers ` { provide: ConfigurationService, useValue: configurationServiceStub }, `

3. You can then start testing the DOM in your spec example:

```
    let newsItem = element.querySelector('li')!;
    expect(newsItem).not.toBeNull();
    expect(newsItem.textContent!.indexOf('We are glad you joined')).toBeGreaterThan(-1);
```

Reply:
Hi @oliverguenther Thank you so much for your help. I tried to complete the test. Because li elements are created dynamically also some elements are added into DOM by conditions, i thought that i should test them after all promises are resolved and i used async() and fixture.whenStable().then().